pkg/kubectl/cmd/edit.go

```diff
@@ -345,16 +345,7 @@ func getMapperAndResult(f cmdutil.Factory, args []string, options *resource.File
 	if err != nil {
 		return nil, nil, nil, "", err
 	}
-	var mapper meta.RESTMapper
-	var typer runtime.ObjectTyper
-	switch editMode {
-	case NormalEditMode:
-		mapper, typer = f.Object()
-	case EditBeforeCreateMode:
-		mapper, typer, err = f.UnstructuredObject()
```

Comment:
> This UnstructuredObject is the direction we want to move.

Yes, I know that. That's why I try to use `UnstructuredObject` in my PR #33250 when implement `create --edit`.

But I recently found that `create --edit` is broken for some resources and we can't let a not working `kubectl create --edit` released.
We have 2 options here:
1) Let them use the same `Object()` which is in PR.
`kubectl edit` and `kubectl create --edit` share most code in `edit.go`. I believe this cahnge won't increase your @deads2k future work.
2) fixes [func ObjectKind()](https://github.com/kubernetes/kubernetes/blob/v1.5.0-beta.0/pkg/client/typed/discovery/unstructured.go#L57).